frontend/src/PublicRoomsPane.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+
+const Row = styled.div`
+  display: table-row;
+  line-height: 23px;
+`;
+const LabelCell = styled.div`
+  display: table-cell;
+  padding-right: 2em;
+  font-weight: bold;
+  width: 125px;
+`;
+const Cell = styled.div`
+  display: table-cell;
+`;
+
+interface RowIProps {
+  roomName: string;
+  numPlayers: number;
+}
+
+const PublicRoomRow = ({ roomName, numPlayers }: RowIProps): JSX.Element => {
+  return (
+    <Row>
+      <Cell>{roomName}</Cell>
+      <Cell>{numPlayers}</Cell>
+    </Row>
+  );
+};
+
+const PublicRoomsPane = (): JSX.Element => {
+  const [publicRooms, setPublicRooms] = useState([]);
+
+  useEffect(() => {
+    loadPublicRooms();
+  }, []);
+  const loadPublicRooms = (): void => {
+    try {
+      const fetchAsync = async (): Promise<void> => {
+        const fetchResult = await fetch("public_games.json");
+        const resultJSON = await fetchResult.json();
+        setPublicRooms(resultJSON);
+      };
+
+      fetchAsync().catch((e) => {
+        console.error(e);
+      });
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return (
+    <div className="">
+      <h3>Public Rooms</h3>
+      <div>
+        <p>
+          The games listed below are open to the public. Join them to find new
+          friends to play with!
+        </p>
+        <p>
+          Copy the room name into the input above, fill out your player name,
```

Comment:
I think we can make these fill it in on click -- we can pass in a `setRoomName` fn as a prop, and then call it in the `onClick` handler for a `PublicRoomRow`. What do you think?